File: Baekjoon/Greedy/20115.py
# https://www.acmicpc.net/problem/20115

import sys
input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().split()))
arr.sort(reverse=True)
res = arr[0]

for i in range(1, len(arr)):
    res += arr[i] / 2

print(res)

File: Baekjoon/Greedy/20115.cpp
// https://www.acmicpc.net/problem/20115

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    vector<double> a;

    cin >> n;
    a.resize(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    sort(a.begin(), a.end());
    double sum = a.back();
    for (int i = 0; i < n - 1; i++) {
        sum += a[i] / 2;
    }
    cout << sum << '\n';

    return 0;
}